File: Data Structures & Algorithms/binary-search/submission-0.java

class Solution {
    
    public int binary_search(int[] nums, int target, int l, int r) {
       if(l>r) return -1;
       int m = l+(r-l)/2;
        if (nums[m] == target) return m;
        return(nums[m]<target)? binary_search(nums,target,m+1,r):
        binary_search(nums,target,l,m-1);
        
    }
    public int search(int nums[],int target){
        return binary_search(nums,target,0,nums.length -1);
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        #[1,2,3,2,2]
        slow_ptr = 0
        fast_ptr = 0

        slow_ptr = nums[0] #1
        fast_ptr =nums[nums[0]] #2

        #phase1
        while slow_ptr != fast_ptr:
            slow_ptr = nums[slow_ptr] #2
            fast_ptr = nums[nums[fast_ptr]] #2

        new_ptr = 0

        while slow_ptr != new_ptr:
            slow_ptr = nums[slow_ptr] # 1 #2
            new_ptr = nums[new_ptr] # 3 # 2

        return new_ptr #2
            